include/wx/private/rescale.h

```diff
@@ -0,0 +1,145 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        wx/private/rescale.h
+// Purpose:     Helpers for rescaling coordinates
+// Author:      Vadim Zeitlin
+// Created:     2021-07-13
+// Copyright:   (c) 2021 Vadim Zeitlin <vadim@wxwidgets.org>
+// Licence:     wxWindows licence
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef _WX_PRIVATE_RESCALE_H_
+#define _WX_PRIVATE_RESCALE_H_
+
+#include "wx/gdicmn.h"
+#include "wx/math.h"
+
+#ifdef __WINDOWS__
+    // Required in order to use wxMulDivInt32().
+    #include "wx/msw/wrapwin.h"
+#endif
+
+// wxRescaleCoord is used to scale the components of the given wxSize by the
+// ratio between 2 scales, with rounding. It doesn't not scale the components
+// if they're set to -1, as this value is special in wxSize.
+//
+// The way it's used is special because we want to ensure there is no confusion
+// between the scale being converted from and the scale being converted to, so
+// instead of just using a single function, we use an intermediate object,
+// which is not supposed to be used directly, but is only returned by From() in
+// order to allow calling To() on it.
+//
+// Another complication is that we want this to work for both wxSize and
+// wxPoint, as well as for just plain coordinate values, so wxRescaleCoord() is
+// an overloaded function and the helper classes are templates, with their
+// template parameter being either wxSize, wxPoint or int.
+
+namespace wxPrivate
+{
+
+template <typename T> class wxRescaleCoordWithValue;
+
+template <typename T>
+class wxRescaleCoordWithFrom
+{
+public:
+    T To(wxSize newScale) const
+    {
+        T value(m_value);
+
+        if ( value.x != -1 )
```

Comment:
Is `wxDefaultCoord` instead of `-1` useful in this file?

Reply:
I guess we should indeed use it. Personally I'm not really sure if `wxDefaultCoord` is more readable than `-1` but we did use it in `ScaleCoordIfSet()`, so there doesn't seem to be any reason not to use it here... I'll update the code, thanks.